Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```diff
@@ -0,0 +1,325 @@
+#nullable enable
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+using Arrowgene.Ddon.Database;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartEquipGradeUpHandler : GameStructurePacketHandler<C2SCraftStartEquipGradeUpReq>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartEquipGradeUpHandler));
+        private static readonly List<StorageType> STORAGE_TYPES = new List<StorageType> {
+            StorageType.ItemBagConsumable, StorageType.ItemBagMaterial, StorageType.ItemBagEquipment, StorageType.ItemBagJob, 
+            StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest
+        };
+        private static readonly List<StorageType> StorageEquipNBox = new List<StorageType> {
+            StorageType.ItemBagEquipment, StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest, StorageType.CharacterEquipment
+        };
+
+        private readonly ItemManager _itemManager;
+        private readonly Random _random;
+
+        public CraftStartEquipGradeUpHandler(DdonGameServer server) : base(server)
+        {
+            _itemManager = Server.ItemManager;
+            _random = new Random();
```

Comment:
No, just using Random.Shared instead of making a new Random instance and calling that instances methods. There must be examples laying around in the server code somewhere